pandas/tseries/period.py

```diff
@@ -281,6 +281,16 @@ def __contains__(self, key):
             return False
         return key.ordinal in self._engine
 
+    def __array_wrap__(self, result, context=None):
```

Comment:
`timedelta64` and `datetime64` are real numpy dtypes, so ufuncs will already choke on them:
```
In [1]: import pandas as pd

In [2]: i = pd.date_range('2000-01-01', periods=5)

In [3]: import numpy as np

In [4]: np.square(i)
TypeError: ufunc 'square' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

In [5]: np.square(i.to_period())
Out[5]:
PeriodIndex(['330671-10-02', '330731-10-03', '330791-10-05', '330851-10-09',
             '330911-10-16'],
            dtype='int64', freq='D')
```